core/src/main/scala/cats/Alternative.scala

```diff
@@ -41,6 +41,30 @@ import simulacrum.typeclass
     (as, bs)
   }
 
+  /**
+   * Separate the inner foldable values into the "lefts" and "rights".
+   * A variant of [[separate]] that is specialized
+   * for Fs that have Foldable instances
+   * which allows for a single-pass implementation
+   * (as opposed to {{{separate}}} which is 2-pass).
+   *
+   * Example:
+   * {{{
+   * scala> import cats.implicits._
+   * scala> val l: List[Either[String, Int]] = List(Right(1), Left("error"))
+   * scala> Alternative[List].separate(l)
+   * res0: (List[String], List[Int]) = (List(error),List(1))
+   * }}}
+   */
+  def separateFoldable[G[_, _], A, B](fgab: F[G[A, B]])(implicit G: Bifoldable[G], FF: Foldable[F]): (F[A], F[B]) =
```

Comment:
I'd prefer for the `Foldable` instance to come first in the list to match the order it appears in the argument, but I guess matching `separate`'s order for consistency is probably more important.